core/storage/btree.rs

```diff
@@ -6387,101 +6387,139 @@ fn page_insert_array(
 /// This function also updates the freeblock list in the page.
 /// Freeblocks are used to keep track of free space in the page,
 /// and are organized as a linked list.
+///
+/// This function may merge the freed cell range into either the next freeblock,
+/// previous freeblock, or both.
 fn free_cell_range(
     page: &mut PageContent,
     mut offset: u16,
     len: u16,
     usable_space: u16,
 ) -> Result<()> {
-    if len < 4 {
-        return_corrupt!("Minimum cell size is 4");
-    }
-
-    if offset > usable_space.saturating_sub(4) {
-        return_corrupt!("Start offset beyond usable space");
-    }
+    const CELL_SIZE_MIN: u16 = 4;
+    turso_assert!(
+        len >= CELL_SIZE_MIN,
+        "free_cell_range: minimum cell size is {CELL_SIZE_MIN}"
+    );
+    turso_assert!(offset <= usable_space.saturating_sub(CELL_SIZE_MIN), "free_cell_range: start offset beyond usable space: offset={offset} usable_space={usable_space}");
```

Comment:
Why assert tho? I remember we change this to return_corrupt! mainly because it emulates what sqlite does.

Reply:
This includes all other corrupt changes

Reply:
Only reason I did it was for easier debugging, can change it back to corrupt